ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,181 @@
+import React, { useContext, useMemo, useRef, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { I18nContext } from '../../../contexts/i18n';
+import { useEqualityCheck } from '../../../hooks/useEqualityCheck';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import { updateViewedNotifications } from '../../../store/actions';
+import { getTranslatedUINoficiations } from '../../../../../shared/notifications';
+import { getSortedNotificationsToShow } from '../../../selectors';
+
+function getActionFunctions() {
+  const actionFunctions = {
+    1: () => {
+      global.platform.openTab({
+        url: 'https://metamask.io/download.html',
+      });
+    },
+    2: () => {
+      global.platform.openTab({
+        url:
+          'https://survey.alchemer.com/s3/6173069/MetaMask-Extension-NPS-January-2021',
+      });
+    },
+    3: () => {
+      global.platform.openTab({
+        url: 'https://community.metamask.io/t/about-the-security-category/72',
+      });
+    },
+  };
+
+  return actionFunctions;
+}
+
+export default function WhatsNewPopup({ onClose }) {
+  const t = useContext(I18nContext);
+
+  const actionFunctions = getActionFunctions();
+
+  const notifications = useSelector(getSortedNotificationsToShow);
+
+  const [seenNotifications, setSeenNotifications] = useState({});
+
+  const popoverRef = useRef();
+
+  const memoizedNotifications = useEqualityCheck(notifications);
+  const idRefMap = useMemo(
+    () =>
+      memoizedNotifications.reduce(
+        (_idRefMap, notification) => ({
+          ..._idRefMap,
+          [notification.id]: React.createRef(),
+        }),
+        {},
+      ),
+    [memoizedNotifications],
+  );
+
+  useEffect(() => {
+    const observer = new window.IntersectionObserver(
+      (entries, _observer) => {
+        entries.forEach((entry) => {
+          if (entry.isIntersecting) {
+            const [id, ref] = Object.entries(idRefMap).find(([_, _ref]) =>
+              _ref.current.isSameNode(entry.target),
+            );
+
+            setSeenNotifications((_seenNotifications) => ({
+              ..._seenNotifications,
+              [id]: true,
+            }));
+
+            _observer.unobserve(ref.current);
+          }
+        });
+      },
+      {
+        root: popoverRef.current,
+        threshold: 1.0,
+      },
+    );
+
+    Object.values(idRefMap).forEach((ref) => {
+      observer.observe(ref.current);
+    });
+
+    return () => {
+      observer?.disconnect();
+    };
+  }, [idRefMap, setSeenNotifications]);
+
+  const renderFirstNotification = (notification, id, date) => {
```

Comment:
Nit: It would be slightly better for performance to lift these two functions out of the render function into the module root, so that they only get declared once rather than once per render.

Reply:
Done in b4849b502